### ISSUE TITLE
question about maxLength

### ISSUE THREAD

**Issue by Marius-Romanus** [KillerCodeMonkey/ngx-quill#1490]
Hello, 

The maxLength attribute, doesn't it prevent the user from continuing to type? what it only does is indicate in the form control that it is invalid, right?

If so, is there a way to get it to continue typing without having to use onContentChanged?

Thank you and best regards.

**Comment by KillerCodeMonkey** [comment]
Just use ngmodel or formcontrol.

Use ngModelChange output of ngmodel to cut the input to your max length or
subscribe to formControls "valueChanges" observable and patch your control
value to only contain your max length.

Marius-Romanus ***@***.***> schrieb am Di., 22. Feb. 2022,
20:31:

**Comment by Marius-Romanus** [comment]
I am subscribed to the event of the ViewChild QuillEditorComponent, I wanted to know if instead of letting the user write, and then delete if it has gone too long, it is possible to directly not let him write.

**Comment by KillerCodeMonkey** [comment]
Yes.... As I described. Use ngmodel or formcontrol and just block user
input if it is too long. So the user is not able to enter something longer
the the max length.

But sure you can even solve this bei using onContentChanged, because you
get the text content there real easy and patch model or control there. So
it is some minor additional logic you need to write.



Marius-Romanus ***@***.***> schrieb am Di., 22. Feb. 2022,
21:56:

**Comment by Marius-Romanus** [comment]
Thanks, I can't. I only manage to delete the characters after they are inserted with quill.setContents, history.undo(), etc. I'll have to research this. If you can give an example I appreciate it, if not, you can close the issue. Thank you very much for the work of the library.

**Comment by KillerCodeMonkey** [comment]
i guess you are using html as format? for this it is really a little bit hard to get it working. The only idea i have is to store the current value of the editor when changed and valid. if not valid use the value before. But this is hard if you have invalid initial value.
if it works for you now, i will close it.
